src/main/java/com/users/exceptions/ControllerExceptionHandler.java

```diff
@@ -0,0 +1,29 @@
+package com.users.exceptions;
+
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.ControllerAdvice;
+import org.springframework.web.bind.annotation.ExceptionHandler;
+import java.time.LocalDateTime;
+
+@ControllerAdvice
```

Comment:
de ce se foloseste aceasta adnotare?

Reply:
pentru a nu repeta logica de tratare a exceptiilor pentru toate metodele @Controller, astfel avem o logica globala pentru ele.

Reply:
Cu alte cuvinte cand se face un call dintr-un controller si pica ceva, intra in aceasta clasa si cauta daca este exceptia aceea tratata.
Noi in task ul asta avem vreo treaba cu controllerele?

Reply:
nu avem treaba aici, dar am avut nevoie pentru a arunca metoda respectiva in Service, acolo am folosit EntityExist si NotFoundException, am inteles la ce te referi, de partea asta avem nevoie la task ul urmator, adica cel de EndPointuri

Reply:
pai ca ai creat exceptia am inteles, dar de clasa asta nu ai nevoie aici, o scoti si o duci la task ul unde ai nevoie